scripts/kg_build.py

```
@@ -199,8 +203,458 @@ class ChunkMetadata:
+        
+        Queries the graph for chunk/document counts and specific anomaly counts (missing embeddings, orphan chunks, checksum mismatches), computes aggregate metrics per-source and totals, compares results against the configured thresholds, writes a sanitized JSON and Markdown QA report to the configured report directory, and records evaluation duration.
+        
+        Returns:
+            QaResult: Aggregated QA evaluation result containing status ("pass" or "fail"), human-readable summary, metrics dictionary, list of detected anomalies, used thresholds, paths to the generated JSON and Markdown reports (relative to the repository when possible), timestamp, report version, and evaluation duration in milliseconds.
+        """
+        eval_start = time.perf_counter()
+        timestamp = datetime.now(timezone.utc)
+        metrics: dict[str, Any] = {}
+        anomalies: list[str] = []
+
+        document_paths = sorted({record.relative_path for record in self._sources})
```

Ruff (0.13.2)

Local variable `document_paths` is assigned to but never used

Remove assignment to unused variable `document_paths`

(F841)